kubernetes/kserve/docker_nightly.py

```diff
@@ -58,9 +58,9 @@
     )
 
     # Push images with latest tag
-    try_and_handle(f"docker push {organization}/{project}:latest-cpu", dry_run)
-    try_and_handle(f"docker push {organization}/{project}:latest-gpu", dry_run)
+    os.system(f"docker push {organization}/{project}:latest-cpu")
```

Comment:
os.system won't error handle so if this line fails it will just keep going

Reply:
Reading more about subprocess, it will execute in parallel to the main process.
Also, if you check the logs here , you will see that it executed the commands after delete was done https://github.com/pytorch/serve/actions/runs/6486104508/job/17613580328

So, not sure whats the right way to handle this.

Reply:
subprocess.call and subprocess.run should be synchronous only subprocess.Popen is async and you make it sync by calling an await

Reply:
Unless, we actually push, I dont have a way to prove that this doesn't work.

But this is what i see in the logs.

![Screenshot 2023-10-11 at 11 36 47 AM](https://github.com/pytorch/serve/assets/16617092/8c21b2f0-6dee-42d6-9909-a6cf3a83467b)

Reply:
I can revert the nightly to use try_and_handle as before and see what happens. But for the official release, we can use the system command. Is that fine?

Reply:
is this the dry run?

Reply:
Yes

Reply:
Reverted the nightly as before. We can see if it works tonight. So far, images were not getting deleted because the prune command didnt have the `-f` option